jax/interpreters/masking.py

```diff
@@ -104,10 +104,15 @@ class ShapeExpr(tuple):  # type ShapeExpr = [Poly]
   def __str__(self):
     return 'ShapeExpr({})'.format(', '.join(map(str, self)))
   def __getitem__(self, idx):
-    if type(idx) is int:
-      return super(ShapeExpr, self).__getitem__(idx)
-    else:
-      return ShapeExpr(super(ShapeExpr, self).__getitem__(idx))
+    poly: Poly = super(ShapeExpr, self).__getitem__(idx)
```

Comment:
Just a note: we can't use variable annotations until we drop Python 3.5 support.